k8s/upgrade-job/src/helm/upgrade.rs

```diff
@@ -88,109 +81,108 @@ impl HelmUpgradeBuilder {
         self
     }
 
-    /// This adds buiilds the HelmUpgrade object.
-    pub(crate) fn build(self) -> Result<HelmUpgrade> {
+    /// This builds the HelmUpgrade object.
+    pub(crate) async fn build(self) -> Result<HelmUpgrade> {
         ensure!(
             self.release_name.is_some() && self.namespace.is_some(),
             HelmUpgradeOptionsAbsent
         );
-
         let release_name = self.release_name.clone().unwrap();
+        let namespace = self.namespace.clone().unwrap();
+
         // Generate HelmReleaseClient.
         let client = HelmReleaseClient::builder()
-            .with_namespace(self.namespace.clone().unwrap())
+            .with_namespace(namespace.clone())
             .build()?;
+
         // Get HelmReleaseElement object for the release specified in CLI options.
         let chart = client.release_info(release_name.clone())?.chart();
 
+        // The version of the Core helm chart (installed as a the parent chart or as a dependent
+        // chart) which is installed in the cluster.
+        let from_version: Version =
+            upgrade::path::version_from_rest_deployment_label(namespace.as_str()).await?;
+
+        // The version of the Core chart which we are (maybe) going to.
+        let chart_dir: PathBuf = self.core_chart_dir.ok_or(
+            NoInputHelmChartDir {
+                chart_name: CORE_CHART_NAME.to_string(),
+            }
+            .build(),
+        )?;
+        let mut chart_yaml_path = chart_dir.clone();
+        chart_yaml_path.push("Chart.yaml");
```

Comment:
```suggestion
        let chart_yaml_path = chart_dir.join("Chart.yaml");
```

Reply:
Also, why not make "core_chart_dir" "core_chart_path"? So we don't even have to append the Chart.yaml

Reply:
We're basically taking the path to the helm chart directory as input, and poking around in multiple files...